src/test/renderWithProviders.tsx

```diff
@@ -122,7 +121,6 @@ export const makeMutationMocks = <T extends (name: keyof AppMutations) => any>(
 
 const defaultQueryMocks: AppQueries = {
   fetchLogo: async () => getLogoMock(),
-  fetchApplicationMetadata: async () => getApplicationMetadataMock(),
```

Comment:
Not quite sure I'm happy now that mocking the application metadata query now needs to be done differently than every other query.. :disappointed:

What's the reason this was needed?

Reply:
It is to move away from the tightly coupled way of setting up a component for testing. Every component should be tested in isolation, which basically is not possible with the current setup. Thus, we don't really have any unit tests, only integration and end-to-end.

Also, `QueryProvider` is not needed as it doesn't actually hold state, but static code, and therefore only adds complexity.  The consequences of the current setup is also that anything we refactor will result in massive changes, because everything is connected to everything. We need to move away from that.  

Without the `QueryProvider`, getting the function to use in `useQuery` also does not need to be a hook, which relives some of the pain you have expressed 😊